webserver/webserver/src/main/java/io/helidon/webserver/WebTracingConfig.java

```diff
@@ -339,12 +341,13 @@ private void doAccept(ServerRequest req, ServerResponse res) {
                     });
         }
 
-        private static class TracingHeaderProvider implements HeaderProvider {
+        static class TracingHeaderProvider implements HeaderProvider {
 
-            private final Map<String, List<String>> headers;
+            private final Map<String, List<String>> headers = new HashMap<>();
 
             TracingHeaderProvider(Map<String, List<String>> headers) {
-                this.headers = headers;
+                // Normalize to lower-case keys, particularly so B3 headers match correctly.
+                headers.forEach((key, values) -> this.headers.put(key.toLowerCase(Locale.ROOT), values));
```

Comment:
We should not normalize on any case - we must use what we get, but treat it correctly as case insensitive